stats/statistics.go

```diff
@@ -0,0 +1,205 @@
+package stats
+
+import (
+	"errors"
+	"expvar"
+	"strconv"
+	"sync"
+)
+
+// errUnexpectedRefCount is the panic used if there is an expected reference counting violation
+var errUnexpectedRefCount = errors.New("unexpected reference counting error")
+
+// The type which is used to implement both the Builder and Statistics interface
+type statistics struct {
+	expvar.Map
+	mu             sync.RWMutex
+	registry       registryClient
+	key            string
+	name           string
+	tags           map[string]string
+	values         *expvar.Map
+	intVars        map[string]*expvar.Int
+	stringVars     map[string]*expvar.String
+	floatVars      map[string]*expvar.Float
+	types          map[string]string
+	built          bool
+	isRecorderOpen bool
+	refsCount      int
+}
+
+func (s *statistics) Key() string {
+	return s.key
+}
+
+func (s *statistics) Name() string {
+	return s.name
+}
+
+func (s *statistics) Tags() map[string]string {
+	return s.tags
+}
+
+func (s *statistics) ValuesMap() *expvar.Map {
+	return s.values
+}
+
+func (s *statistics) Values() map[string]interface{} {
+	values := make(map[string]interface{})
+	n := s.ValuesMap()
+	n.Do(func(kv expvar.KeyValue) {
+		var f interface{}
+		var err error
+		switch v := kv.Value.(type) {
+		case *expvar.Float:
+			f, err = strconv.ParseFloat(v.String(), 64)
```

Comment:
You can collapse these `if` statements and following `err` checks to three lines instead of four.